src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Block.cs

```diff
@@ -48,5 +60,47 @@ public override BoundNode VisitNoOpStatement(BoundNoOpStatement node)
                 ? new BoundBlock(node.Syntax, ImmutableArray<LocalSymbol>.Empty, ImmutableArray<BoundStatement>.Empty)
                 : _instrumenter.InstrumentNoOpStatement(node, node);
         }
+
+        private void LowerPendingUsingDeclarations(ArrayBuilder<BoundStatement> statements, ImmutableArray<int> unloweredDeclarationIndicies)
+        {
+            // work backwards and lower the using declarations
```

Comment:
Working backwards originally seemed like a good idea to me, but I think doing as much natural recursion as possible would be even simpler. The advantage of doing it this way is avoiding blowing the stack, but I'm not sure how much it's worth it.